controllers/flowlogspipeline/flp_objects.go

```diff
@@ -400,17 +414,17 @@ func (b *builder) autoScaler() *ascv2.HorizontalPodAutoscaler {
 //+kubebuilder:rbac:groups=autoscaling,resources=horizontalpodautoscalers,verbs=create;delete;patch;update;get;watch;list
 //+kubebuilder:rbac:groups=core,resources=pods;services;nodes,verbs=get;list;watch
 
-func buildAppLabel() map[string]string {
+func buildAppLabel(confKind string) map[string]string {
 	return map[string]string{
-		"app": constants.FLPName,
+		"app": constants.FLPName + FlpConfSuffix[confKind],
 	}
 }
 
 func buildClusterRole() *rbacv1.ClusterRole {
 	return &rbacv1.ClusterRole{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:   constants.FLPName,
-			Labels: buildAppLabel(),
+			Labels: buildAppLabel(""),
 		},
 		Rules: []rbacv1.PolicyRule{{
 			APIGroups: []string{""},
```

Comment:
On cluster roles, actually, they need to be split, they are not the same for ingester and for transformer.
Only the transformer needs to watch pods/services/nodes/replicasets
And I guess only the ingester needs to use hostnetwork (I'm actually not sure why we need that, @mariomac can you remind me?)

We can create two clusterrole objects (even in the case of a single FLP if it makes it simpler), and deal with cluster role attribution via clusterrolebindings

Reply:
Also I don't know why we need horizontalpodautoscalers here: it's not FLP itself who create the HPA ?

Reply:
As far as I know, `hostNetwork: true` is only required by the eBPF agent, so as long as it should be associated with another cluster role, the hostnetwork label here shouldn't be necessary.

Reply:
Doesn't we need host network for daemonset deployment? So we can bind node port which is used by ovs?

Reply:
We can do a try removing it, see if it works... Mario you're the one who added it :) (cf #29 )

Reply:
For the record here is the error message without hostNetwork permission:
Forbidden: not usable by user or serviceaccount, spec.containers[0].securityContext.containers[0].hostPort: Invalid value: 2055: Host ports are not allowed to be used, provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]

Reply:
btw the way, I'm dealing with issues with `hostNetwork` when running with KIND instead of openshift, for this PR https://github.com/netobserv/network-observability-operator/pull/97
Obviously this `hostnetwork` config is specific to openshift and we need to do the equivalent with vanilla kube .. sounds a bit similar to what @mariomac is doing with the ebpf agent.

Without it, FLP never receives the ipfix flows. If I add `hostNetwork: true` on FLP pods, then they start receiving the flows ... but then they're unable to reach Loki, I guess because Loki is on the "cluster" network, not the host network 😭
Anyone got an idea?

Reply:
In openshift without this permission, the pod is not allowed to be created.

Here you are saying that the pod is created, is able to bind the host port, but does not receive ovs flows without the permission? This is strange...

Reply:
Maybe the error is because you removed the permission to use hostnetwork and did not remove the "hostnetwork" specification from the FLP daemonset. Here is an example of FLP running as daemonset and not requiring hostnetwork (only the agent requires it): https://github.com/netobserv/netobserv-ebpf-agent/blob/main/deployments/flp-daemonset.yml

Reply:
@jotak maybe changing the DNS policy fixes the problem:
https://github.com/netobserv/netobserv-ebpf-agent/blob/main/deployments/flp-daemonset.yml#L18-L20
However I still think FLP shouldn't need the `hostnetwork: true` tag. In the linked deployment, FLP works without requiring it, both in OpenShift and Kubernetes.

Reply:
> Maybe the error is because you removed the permission to use hostnetwork and did not remove the "hostnetwork" specification from the FLP daemonset.

Yes but the daemonset needs to bind the node port, this is how ovs reach it in daemonset mode :
https://github.com/netobserv/network-observability-operator/blob/d0a45a19c287e681e063483d1991d050b8573896/controllers/ovs/flowsconfig_reconciler.go#L123

Reply:
I am not saying that FLP as a DaemonSet would not work.

But if you deploy FLP as a DaemonSet using container port like in your example then you need a service to make this port reachable. And if you do so, then you can not guarantee anymore that ovs flows will be send to the corresponding FLP container on the same node hence avoiding network communications.

To me, the point of using DaemonSet and host network instead of Deployment and service was to grantee that the communication between ovs and flp would be kept on the same node.

Reply:
Yeah, but please observe that in the example configuration, the eBPF agent is still sending to the FLP in its own host, that bounds the port to a host port:
https://github.com/netobserv/netobserv-ebpf-agent/pull/27/files#diff-d67b1319dcea073fb864a6f7d9e93ac5d56d34d531779dcf092e39364c72318dR31

Reply:
I am missing something in your example then:
- ebpf-agent is sending to status.hostIP which correspond to the node IP and to the port 9999 
- flp is binding the container port 9999

I am missing the part in your example that bind the host port 9999 to the container port 9999.

My understanding is that the flp daemonset would also require hostNetwork to true so the container port getting mapped is the host port.

Reply:
Yes, I know how hostNetwork feature works.

> it will be automatically bound to the host port 9999 unless you explicitly set another hostPort property.

I am very surprise about this.

Do you have documentation about it? What kind of cluster are you using for this test? The [k8s official documentation](https://kubernetes.io/docs/concepts/services-networking/connect-applications-service/#exposing-pods-to-the-cluster) does not mention surch automatic bind and at the oposite state the following:
> Note that the containers are not using port 80 on the node, nor are there any special NAT rules to route traffic to the pod.

Reply:
For more safety, I'd specify `hostPort` to the same value as `containerPort` but it still should work without setting `hostNetwork: true`

Reply:
JIRA created, thanks for clearing that up! https://issues.redhat.com/browse/NETOBSERV-362